test/functional/utils/test-rate-limit.js

```diff
@@ -65,4 +65,22 @@ describe('rate-limit', () => {
     expect(callback).to.be.calledOnce;
     expect(callback).to.be.calledWith(7, 'another param');
   });
+
+  it('is re-entrant', () => {
+    let calls = 0;
+    let rateLimitedCallback;
+    function fn(call) {
+      if (calls++ < 1) {
+        rateLimitedCallback(calls);
```

Comment:
It says:

1. rate limited function is not called repeatedly ([Line](https://github.com/ampproject/amphtml/pull/8867/files#diff-36efe2fad9c166727e5f020c1a069d02R76))
  - `rateLimit(funciton recursive() { recursive(); })` would have caused a stack overflow
2. rate limited calls that happen _during_ `fire` execution will run after the timeout. ([Line](https://github.com/ampproject/amphtml/pull/8867/files#diff-36efe2fad9c166727e5f020c1a069d02R84))